src/main.ts

```diff
@@ -1,16 +1,20 @@
 import { POSTCODE_REGEXES } from './postcode-regexes';
 
-export { CountryCode } from './postcode-types';
+import { CountryCode } from './postcode-types';
 
-export const postcodeValidator = (postcode: string, country: string): boolean => {
-  if (!POSTCODE_REGEXES.has(country)) {
+export type OtherStringOr<T extends string> = T | (string & {});
```

Comment:
Firstly, thanks for making these changes. 

Could we not just use the `CountryCode` type directly as the function throws on unrecognised country codes?